src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```diff
@@ -880,110 +709,14 @@ PAGINATION_INDEX_PLACEHOLDER, getPaginationType()),
       }
     }
 
-    // Validate OAuth2 properties
-    if (!containsMacro(PROPERTY_OAUTH2_ENABLED) && this.getOauth2Enabled()) {
-      String reasonOauth2 = "OAuth2 is enabled";
-      assertIsSet(getAuthUrl(), PROPERTY_AUTH_URL, reasonOauth2);
-      assertIsSet(getTokenUrl(), PROPERTY_TOKEN_URL, reasonOauth2);
-      assertIsSet(getClientId(), PROPERTY_CLIENT_ID, reasonOauth2);
-      assertIsSet(getClientSecret(), PROPERTY_CLIENT_SECRET, reasonOauth2);
-      assertIsSet(getRefreshToken(), PROPERTY_REFRESH_TOKEN, reasonOauth2);
-    }
-    // Validate Authentication properties
-    AuthType authType = getAuthType();
-    switch (authType) {
-      case OAUTH2:
-        String reasonOauth2 = "OAuth2 is enabled";
-        if (!containsMacro(PROPERTY_AUTH_URL)) {
-          assertIsSet(getAuthUrl(), PROPERTY_AUTH_URL, reasonOauth2);
-        }
-        if (!containsMacro(PROPERTY_TOKEN_URL)) {
-          assertIsSet(getTokenUrl(), PROPERTY_TOKEN_URL, reasonOauth2);
-        }
-        if (!containsMacro(PROPERTY_CLIENT_ID)) {
-          assertIsSet(getClientId(), PROPERTY_CLIENT_ID, reasonOauth2);
-        }
-        if (!containsMacro((PROPERTY_CLIENT_SECRET))) {
-          assertIsSet(getClientSecret(), PROPERTY_CLIENT_SECRET, reasonOauth2);
-        }
-        if (!containsMacro(PROPERTY_REFRESH_TOKEN)) {
-          assertIsSet(getRefreshToken(), PROPERTY_REFRESH_TOKEN, reasonOauth2);
-        }
-        break;
-      case SERVICE_ACCOUNT:
-        String reasonSA = "Service Account is enabled";
-        assertIsSet(getServiceAccountType(), PROPERTY_NAME_SERVICE_ACCOUNT_TYPE, reasonSA);
-        boolean propertiesAreValid = validateServiceAccount(failureCollector);
-        if (propertiesAreValid) {
-          try {
-            String accessToken = OAuthUtil.getAccessTokenByServiceAccount(this);
-          } catch (Exception e) {
-            failureCollector.addFailure("Unable to authenticate given service account info. ",
-                                        "Please make sure all infomation entered correctly")
-              .withStacktrace(e.getStackTrace());
-          }
-        }
-        break;
-      case BASIC_AUTH:
-        String reasonBasicAuth = "Basic Authentication is enabled";
-        if (!containsMacro(PROPERTY_USERNAME)) {
-          assertIsSet(getUsername(), PROPERTY_USERNAME, reasonBasicAuth);
-        }
-        if (!containsMacro(PROPERTY_PASSWORD)) {
-          assertIsSet(getPassword(), PROPERTY_PASSWORD, reasonBasicAuth);
-        }
-        break;
-    }
+    validateOAuth(failureCollector);
```

Comment:
Implement validate method in the parent class instead, and move the logic there, You can call super.validate from this method instead.